pkg/monitor/intervalcreation/e2etest.go

```diff
@@ -30,47 +29,37 @@ func IntervalsFromEvents_E2ETests(events monitorapi.Intervals, _ monitorapi.Reso
 			from = lastStart
 		}
 		level := monitorapi.Info
-		endState := "MISSING"
-		switch {
-		case strings.Contains(event.Message, "finishedStatus/Flaked"):
+		switch testStatus {
+		case "Flaked":
 			level = monitorapi.Warning
-			endState = "Flaked"
-		case strings.Contains(event.Message, "finishedStatus/Failed"):
+		case "Failed":
 			level = monitorapi.Error
-			endState = "Failed"
-		case strings.Contains(event.Message, "finishedStatus/Skipped"):
+		case "Skipped":
 			level = monitorapi.Info
-			endState = "Skipped"
-		case strings.Contains(event.Message, "finishedStatus/Passed"):
+		case "Passed":
 			level = monitorapi.Info
-			endState = "Passed"
-		case strings.Contains(event.Message, "finishedStatus/Unknown"):
+		case "Unknown":
+			level = monitorapi.Warning
+		default:
 			level = monitorapi.Warning
-			endState = "Unknown"
 		}
 
 		delete(testNameToLastStart, testName)
-		ret = append(ret, monitorapi.Interval{
-			Condition: monitorapi.Condition{
-				Level:   level,
-				Locator: event.Locator,
-				Message: fmt.Sprintf("e2e test finished As %q", endState),
-			},
-			From: from,
-			To:   event.From,
-		})
+		ret = append(ret, monitorapi.NewInterval(monitorapi.SourceE2ETest, level).Locator(event.StructuredLocator).
+			Message(monitorapi.NewMessage().
+				HumanMessagef("e2e test finished as %q", testStatus).
+				WithAnnotation(monitorapi.AnnotationStatus, testStatus)).
+			From(from).To(event.From).Build())
```

Comment:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/28099/pull-ci-openshift-origin-master-e2e-aws-ovn-fips/1684981924319203328/artifacts/e2e-aws-ovn-fips/openshift-e2e-test/artifacts/junit/e2e-events_20230728-183147.json has them, so they must be there.

I'm not sure why they're not in the chart, it looks like it might be pre-existing, from another PR: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28097/pull-ci-openshift-origin-master-e2e-aws-ovn-fips/1685991199694917632

The code is doing:

```javascript
    function isE2EFailed(eventInterval) {
        if (eventInterval.locator.includes("e2e-test/") && eventInterval.message.includes("finished As \"Failed")) {
            return true
        }
        return false
    }

    function isE2EFlaked(eventInterval) {
        if (eventInterval.locator.includes("e2e-test/") && eventInterval.message.includes("finished As \"Flaked")) {
            return true
        }
        return false
    }

    function isE2EPassed(eventInterval) {
        if (eventInterval.locator.includes("e2e-test/") && eventInterval.message.includes("finished As \"Passed")) {
            return true
        }
        return false
    }
```

Which seems like it should still work...

And then there's: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/openshift-kubernetes-1646-nightly-4.14-e2e-aws-sdn-upgrade/1684997334510866432 from a recent payload test in the kube repo. I think something is broken, but it may not be this PR, checking more...

Reply:
I see the E2ETest types now.  I think I see why they don't show up in the timeline.

  
    function isE2EPassed(eventInterval) {
        if (eventInterval.locator.includes("e2e-test/") && eventInterval.message.includes("finished As \"Passed")) {
            return true
        }
        return false
    }
    
Looks like the function wants `As` but the data has `as`    
    
          {
            "level": "Info",
            "locator": "e2e-test/\"[sig-node] ConfigMap should run through a ConfigMap lifecycle [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]\"",
            "message": "status/Passed e2e test finished as \"Passed\"",
            "tempStructuredLocator": {
                "type": "E2ETest",
                "keys": {
                    "e2e-test": "[sig-node] ConfigMap should run through a ConfigMap lifecycle [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]"
                }
            },
            "tempStructuredMessage": {
                "reason": "",
                "cause": "",
                "humanMessage": "e2e test finished as \"Passed\"",
                "annotations": {
                    "status": "Passed"
                }
            },
            "from": "2023-07-31T17:30:32Z",
            "to": "2023-07-31T17:30:44Z"
        },
        
        
May not be this change but could be worth fixing (now or later)

Reply:
Eh, well I guess it is changed in this PR

Reply:
Oof, yes that's it. Thanks @neisw

Reply:
Taking it back to finished As, as this spans multiple html files and sippy. We'll have structured filtering once this work is all done and can stop doing things like this to sort out the events we want.